packages/xcodeml-tools/package.py

```diff
@@ -0,0 +1,70 @@
+__author__      = "Mikhail Zhigun"
+__copyright__   = "Copyright 2020, MeteoSwiss"
+
+from spack import *
+import sys
+assert sys.version_info[0] >= 3 and sys.version_info[1] >= 6, 'Python >= 3.6 is required'
+import os, subprocess
+
+_GIT = 'https://github.com/MeteoSwiss-APN/xcodeml-tools.git'
+
+
+def _get_latest_commit_id() -> str:
+    git_url = _GIT
+    args = ['git', 'ls-remote', git_url, 'HEAD']
+    timeout = 5  # seconds
+    p = subprocess.run(args=args, timeout=timeout, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    s = p.stdout.decode('utf-8')
+    assert p.returncode == 0, f'Failed to connect to {git_url} error: {s}'
+    commit = s.strip().split()[0]
+    return commit
+
+class XcodemlTools(AutotoolsPackage):
+    """Set of tools for translating C and Fortran code to XCodeML and back """
+
+    homepage = 'https://omni-compiler.org/manual/en/'
+    git = _GIT
+
+    maintainers = ['FrostyMike']
+
+    version('latest', branch='master', commit=_get_latest_commit_id())
+    version('test', branch='master', commit='826403a')
+    #version('test', branch='master', commit='39edc71')
+    
+    depends_on('autoconf@2.69:')
+    depends_on('m4')
+    depends_on('automake')
+    depends_on('libxml2%gcc')
+    depends_on('java@8:')
+    depends_on('bison%gcc')
+    depends_on('flex%gcc')
+
+    def configure_args(self):
+        args = ['--prefix=' + self.prefix,
+                '--with-libxml2=' + self.spec['libxml2'].prefix,
+                '--without-native-fortran-compiler']
+        java_prefix = self.spec['java'].prefix
+        path = {'java': '/bin/java',
+                'javac': '/bin/javac',
+                'jar': '/bin/jar'}
+        for name, rel_path in path.items():
+            abs_path = os.path.normpath(os.path.join(java_prefix, rel_path))
+            assert os.path.exists(abs_path) and os.path.isfile(abs_path), '%s not found at "%s"' % (name, abs_path)
+            path[name] = abs_path
```

Comment:
Might be a little overkill since spack should already check that your java dependency does exist but ok

Reply:
I like to play it safe. It does check that dependency is there, but does not say wher eindividual files are (java, javac, jar). Since I construct the path myself(e.g. prefix + bin/java ), I double-check if it was done correctly.